README.md

```diff
@@ -22,21 +22,23 @@ go get github.com/ory/graceful
 ```go
 package main
 
-import "github.com/ory/graceful"
-import "net/http"
+import (
+    "net/http"
+    "log"
+
+    "github.com/ory/graceful"
+)
 
 func main() {
-    server := graceful.PatchHTTPServerWithCloudflareConfig(&http.Server{
+    server := graceful.WithDefaults(&http.Server{
         Addr: ":54932",
         // Handler: someHandler,
     })
 
-    if err := server.Graceful(func () {
-        if err := server.ListenAndServe(); err != nil {
-            // ...
-        }
-    }); err != nil {
-        // ...
+    log.Println("main: Starting the server")
+    if err := graceful.Graceful(server.ListenAndServe, server.Shutdown); err != nil {
```

Comment:
Cool idea! I got thrown off at first because I thought this won't work with [`ListenAndServeTLS(certFile, keyFile string) error`](https://golang.org/pkg/net/http/#Server.ListenAndServeTLS) but then I realized that the signature is the same `func() error` as before, so we can still provide a lambda here right?

Maybe I'll add an example for TLS as well.